packages/dart/npt_flutter/lib/features/profile/bloc/profile_bloc.dart

```diff
@@ -158,26 +162,20 @@ class ProfileBloc extends LoggingBloc<ProfileEvent, ProfileState> {
         emit(ProfileStarting(uuid, profile: profile, status: err));
       });
 
-      cancelSubs = () {
+      cancel = () {
         progressSub?.cancel();
         progressSub = null;
 
         errorSub?.cancel();
         errorSub = null;
+
+        if (sc is SocketConnector) sc.close();
```

Comment:
renamed and added an extra attempt to close the socket connector, usually happens when an error state occurs.